packages/http-signature-utils/src/utils/headers.ts

```diff
@@ -0,0 +1,50 @@
+import { createContentDigestHeader } from 'httpbis-digest-headers'
+import {
+  createSignatureHeaders,
+  SignatureHeaders,
+  SignOptions
+} from './signatures'
+
+interface ContentHeaders {
+  'Content-Digest': string
+  'Content-Length': string
+  'Content-Type': string
+}
+
+export interface Headers extends SignatureHeaders, Partial<ContentHeaders> {}
+
+const createContentHeaders = (body: string): ContentHeaders => {
+  return {
+    'Content-Digest': createContentDigestHeader(
+      JSON.stringify(JSON.parse(body)),
+      ['sha-512']
+    ),
+    'Content-Length': Buffer.from(body as string, 'utf-8').length.toString(),
+    'Content-Type': 'application/json'
+  }
+}
+
+export const createHeaders = async ({
+  request,
+  privateKey,
+  keyId
+}: SignOptions): Promise<Headers> => {
```

Comment:
Could generic `SignOptions` and `Headers` types be used to enforce that a request with `body` will return headers with content headers?

Reply:
I don't understand

Reply:
Not sure if it's worth it
```
type NoBodyRequest = SignOptions['request'] & {
  body?: never
}

type NoBodyOptions = Omit<SignOptions, 'request'> & {
  request: NoBodyRequest
}

type NoBodyHeaders = SignatureHeaders

type BodyRequest = SignOptions['request'] & {
  body: string | Buffer
}

type BodyOptions = Omit<SignOptions, 'request'> & {
  request: BodyRequest
}

type BodyHeaders = SignatureHeaders & ContentHeaders

export const createHeaders:
  | ((options: NoBodyOptions) => Promise<NoBodyHeaders>)
  | ((options: BodyOptions) => Promise<BodyHeaders>) = async ({
  request,
  privateKey,
  keyId
}) => {
```